### PR TITLE
Document Lighting v3

### PR DESCRIPTION
**Pull request by mcendu**
To consider:
- [x] Which parts fall into basic lighting, and which is intermediate/advanced?

### CODE REVIEW

**Comment by bloodcloak** [comment]
image
this is the current breakdown of the pages. So I guess with that in mind maybe overview of the system in basic with all of the more vanilla technical settings in intermediate? Then rest in advanced.

File: wiki/mapping/intermediate-lighting.md [hunk 1]
```diff
@@ -17,7 +17,127 @@ You've got your lighting feet wet in [Basic Lighting](./basic-lighting.md) and h
...
+segments are grouped into chunks: section 2 affects the lights right
+after section 1, section 3 affects those right after section 2, and
+so on. The section controlled is given in the Id field.
+
+In step and offset mode, one chunk would be affected by the lane for
+every step. The size of a step is settable. If step is set to 2, then
+the 1st chunk would be controlled by the lane, the 2nd in turn would
+not, the 3rd would, the 4th would not, and so on. If it is 3 instead,
+the lane would control chunks 1, 3, 7, 10 and on. The selection do not
+need to start at the 1st chunk; the offset, titled <q>Light #</q>, is
```
Comment (bloodcloak) [inline wiki/mapping/intermediate-lighting.md:63]: Linting error blocks. replace the <q> and </q> with ` to highlight/differentiate or just regular "

File: wiki/mapping/intermediate-lighting.md [hunk 2]
```diff
@@ -17,7 +17,127 @@ You've got your lighting feet wet in [Basic Lighting](./basic-lighting.md) and h
...
+There are two modes, wave and step, the meaning being similar to that
+for duration. With wave distribution, the number specifies the final
+intensity or rotation of the last chunk in order compared to that of
+the event itself. With step distribution, the number instead stands
+for the difference between a chunk and the next.
+
+Normally, the first event on the lane would not be affected by
+distribution; instead, the event sets all chunks' rotation or
+brightness to be the same. This is sometimes useful. When this
+is not desired, <q>Affect First</q> can be checked to make
```
Comment (bloodcloak) [inline wiki/mapping/intermediate-lighting.md:127]: See suggestion to replace above

File: wiki/mapping/basic-lighting.md [hunk 3]
```diff
@@ -38,21 +39,92 @@ though they may be in slightly different positions or not present in some cases.
...
   On uses the second (boost) pair of colors. There is no fade or flash option for this event.
 
-## Lighting Events
-Each lighting type can make use of four different lighting events.
-
-* **On:** Turns a light on that will stay on until you place a different event type.
-* **Off:** Turning a light off can have as much impact as turning it on, or more.
-* **Flash:** Briefly flashes brighter than a standard "on" event then stays on until you place a different event type.
-* **Fade:** Briefly flashes brighter than a standard "on" event then fades off after a couple of seconds.
+In Weave and later environments, or v3 environments, the set of objects
```
Comment (bloodcloak) [inline wiki/mapping/basic-lighting.md:42]: `In Weave and later environments, or v3 environments,` suggest `In environments that use the v3 Group Lighting System (Weave and later),`

**Review by bloodcloak — CHANGES_REQUESTED** [review]
The double spaces after the end of a line is what tells the site renderer to create a new line. Instead of joining the two adjacent together. 

Should not remove this double space in
Basic Lighting Lines: 55, and 241
Intermediate Lighting Lines 108, 127, 160, and 167

**Comment by mcendu** [comment]
I have no access to my computer for the time being, so it might take some time before I can actually apply the suggestions.

On the lint error: I have to admit that the q element is not appropriately used here; per WHATWG HTML standard, this element cannot be used when it does not represent a quote. Would replace with &ldquo; and &rdquo; pairs once I am back.

> Should not remove this double space in
Basic Lighting Lines: 55, and 241
Intermediate Lighting Lines 108, 127, 160, and 167

My editor is set to automatically trim trailing whitespaces. While I personally prefer to use <br> for this purpose, using EditorConfig to tell the editor to not trim trailing whitespaces might better fit our markup formatting style.

> In Weave and later environments, or v3 environments, suggest In environments that use the v3 Group Lighting System (Weave and later),

Agreed.

**Comment by mcendu** [comment]
P.S. The EditorConfig actually already tells the editor to not trim whitespaces. Something is wrong with my configuration…
________________________________
发件人: Nathan Du ***@***.***>
发送时间: Sunday, May 7, 2023 4:16:03 PM
收件人: bsmg/wiki ***@***.***>
主题: Re: [bsmg/wiki] Document Lighting v3 (PR #497)

I have no access to my computer for the time being, so it might take some time before I can actually apply the suggestions.

On the lint error: I have to admit that the q element is not appropriately used here; per WHATWG HTML standard, this element cannot be used when it does not represent a quote. Would replace with &ldquo; and &rdquo; pairs once I am back.

> Should not remove this double space in
Basic Lighting Lines: 55, and 241
Intermediate Lighting Lines 108, 127, 160, and 167

My editor is set to automatically trim trailing whitespaces. While I personally prefer to use <br> for this purpose, using EditorConfig to tell the editor to not trim trailing whitespaces might better fit our markup formatting style.

> In Weave and later environments, or v3 environments, suggest In environments that use the v3 Group Lighting System (Weave and later),

Agreed.

**Comment by XAce1337manX** [comment]
🙏 The table reformat. I can add preview images for the environments later.

Wonder if we should move the table to another page considering it takes up over half the page now.

**Review by bloodcloak — APPROVED** [review]

**Pull request merged**